linchpin/provision/roles/aws/library/auth_driver.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
```

Comment:
we usually use symbolic link to library, change of policy?

Reply:
What do you mean?

Reply:
well usually inside the roles we have a symbolic link to the library, in this case it would be
linchpin/provision/roles/aws/library -> ../../library

Reply:
I can put them symbolic link back in.  The reason I did this is to separate out the files in library that are actually necessary for each provider so that we can more easily separate each provider into repos when that becomes an option.  I can also link the individual files instead of the entire library directory if you think that's a more elegant solution.  That way we know the true dependencies of each role

Reply:
Ideally, this would be the situation:
* Each role has its own GitHub repo so that we can store the roles on Galaxy
* `library` and `filter_plugins`/`FilterUtils` are their own separate repos which are included in the roles through the use of a submodule

But that's far off at best

Reply:
I think symbolic link expresses that ideal and will be a reminder when the role will be moved. The part I'm worried of is that some changes will be specific for AWS and it will be hard to reunite as time go by. But I would like to track this issue for future work of OASIS and Linchpin, maybe some of the library and plugins could be moved to specific roles (for example Beaker) and other parts could become generic solutions (authentication).